commit 457620c99f806f19d8475497a99200e741b72b1a (refs/changes/63/1063/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-12-05T15:19:26-08:00 (2 years, 10 months ago)
    
    joyent/node-mname#11 name compression offset limits are too high

diff --git a/lib/dns-buffer.js b/lib/dns-buffer.js
index 41c0cfd..5e45df4 100644
--- a/lib/dns-buffer.js
+++ b/lib/dns-buffer.js
@@ -135,8 +135,8 @@ DNSBuffer.prototype.writeName = function (name) {
 
         var rlen;
         while (rparts.length > 0) {
-                /* Can only use ptrs to things in the first 0xc000 bytes. */
-                if (this._offset < 0xc000)
+                /* Can only use ptrs to things in the first 0x3fff bytes. */
+                if (this._offset <= 0x3fff)
                         this._ncache.add(parts, this._offset);
 
                 var part = rparts.shift();
@@ -150,7 +150,7 @@ DNSBuffer.prototype.writeName = function (name) {
         }
 
         if (suffix.offset !== undefined) {
-                assert.ok(suffix.offset < 0xc000);
+                assert.ok(suffix.offset <= 0x3fff);
 
                 var ptr = suffix.offset & 0xff;
                 rlen = NAME_PTR | ((suffix.offset & 0x3f00) >> 8);
diff --git a/package.json b/package.json
index b0ec040..0025677 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "mname",
   "description": "DNS server library for node.js",
-  "version": "1.3.0",
+  "version": "1.3.1",
   "author": "arekinath <alex@cooperi.net>",
   "contributors": [
     "Mark Cavage",
